cmd/ncproxy/server.go

```diff
@@ -21,9 +21,14 @@ type server struct {
 	ttrpc *ttrpc.Server
 	grpc  *grpc.Server
 	conf  *config
+	db    *bolt.DB
 }
 
-func newServer(ctx context.Context, conf *config) (*server, error) {
+func newServer(ctx context.Context, conf *config, dbPath string) (*server, error) {
+	db, err := bolt.Open(dbPath, 0600, nil)
```

Comment:
Do we need to close this db somewhere?

Reply:
I could include it in `gracefulShutdown` but I'm not sure I understand what it provides us. From reading [the doc on Close](https://pkg.go.dev/github.com/boltdb/bolt#DB.Close) it looks like it releases all database resources. Not sure if that means something would be leaked without closing it or if it means that when the process with the open database closes then everything will be cleaned up and we don't need to worry about always closing it.

Reply:
I think of this in terms of encapsulation. We ideally want the `server` object to be self contained and not leave resources lying around when it is cleaned up. So if `server` opens these resources, then we should probably ensure they are cleaned up when it is stopped. Given that there is already a `gracefulShutdown` function on `server`, it seems we already expect to be able to cleanly release any resources held by `server`, so I think this should be included as well.

For the specific use case we have today, you are correct that we could just wait for process termination which will release these resources. If we really want to rely on that then we should probably remove `gracefulShutdown` completely, as while it exists it gives the impression that we support shutting down the server without terminating the process.

Reply:
That's fair, I'll add it to `gracefulShutdown`.